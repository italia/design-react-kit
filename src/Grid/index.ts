export { GridItem } from './GridItem';
export { GridItemText } from './GridItemText';
export { GridItemTextWrapper } from './GridItemTextWrapper';
export { GridList } from './GridList';
export { GridRow } from './GridRow';

export type { GridItemProps } from './GridItem';
export type { GridItemTextProps } from './GridItemText';
export type { GridItemTextWrapperProps } from './GridItemTextWrapper';
export type { GridListProps } from './GridList';
export type { GridRowProps } from './GridRow';
