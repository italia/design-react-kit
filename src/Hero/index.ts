export { Hero } from './Hero';
export { HeroBackground } from './HeroBackground';
export { HeroBody } from './HeroBody';
export { HeroButton } from './HeroButton';
export { HeroCategory } from './HeroCategory';
export { HeroTitle } from './HeroTitle';

export type { HeroProps } from './Hero';
export type { HeroBackgroundProps } from './HeroBackground';
export type { HeroBodyProps } from './HeroBody';
export type { HeroButtonProps } from './HeroButton';
export type { HeroCategoryProps } from './HeroCategory';
export type { HeroTitleProps } from './HeroTitle';
