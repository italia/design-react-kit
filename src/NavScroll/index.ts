/*
 * This work derives from the React Use Navscroll library
 * Released under the MIT license by Marco Liberati
 * Code: https://github.com/dej611/react-use-navscroll
 */
/* @internal */
export type { ChangesType, RegisterOptions, useNavScrollArgs, useNavScrollResult } from './types';

export { useNavScroll } from './useNavScroll';
