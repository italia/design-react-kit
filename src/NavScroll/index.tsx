export type { ChangesType, RegisterOptions, useNavScrollArgs, useNavScrollResult } from './types';

export { useNavScroll } from './useNavScroll';
