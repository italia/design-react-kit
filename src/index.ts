export {
  BreadcrumbItem,
  ButtonGroup,
  ButtonToolbar,
  CardColumns,
  CardDeck,
  CardFooter,
  CardGroup,
  CardHeader,
  CardImg,
  CardImgOverlay,
  CardLink,
  CardSubtitle,
  Col,
  Container,
  Fade,
  Form,
  FormFeedback,
  FormGroup,
  FormText,
  InputGroup,
  InputGroupText,
  Label,
  ListGroup,
  ListGroupItem,
  ListGroupItemHeading,
  ListGroupItemText,
  Media,
  NavItem,
  NavLink,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  PaginationItem as PagerItem,
  PaginationLink as PagerLink,
  Popover,
  PopoverBody,
  PopoverHeader,
  Row,
  TabContent,
  TabPane,
  Table,
  Tooltip,
  UncontrolledAlert,
  UncontrolledCollapse,
  UncontrolledTooltip,
  Util
} from 'reactstrap';

export { Accordion } from './Accordion/Accordion';
export { AccordionBody } from './Accordion/AccordionBody';
export { AccordionHeader } from './Accordion/AccordionHeader';
export { AccordionItem } from './Accordion/AccordionItem';
export { Alert } from './Alert/Alert';
export { AvatarContainer } from './Avatar/AvatarContainer';
export { AvatarExtraText } from './Avatar/AvatarExtraText';
export { AvatarGroupContainer } from './Avatar/AvatarGroupContainer';
export { AvatarIcon } from './Avatar/AvatarIcon';
export { AvatarPresence } from './Avatar/AvatarPresence';
export { AvatarStatus } from './Avatar/AvatarStatus';
export { AvatarWrapper } from './Avatar/AvatarWrapper';
export { BackToTop } from './BackToTop/BackToTop';
export { Badge } from './Badge/Badge';
export { BottomNav } from './BottomNav/BottomNav';
export { BottomNavItem } from './BottomNav/BottomNavItem';
export { Breadcrumb } from './Breadcrumb/Breadcrumb';
export { Button } from './Button/Button';
export { Callout } from './Callout/Callout';
export { CalloutMoreFooter } from './Callout/CalloutMoreFooter';
export { CalloutText } from './Callout/CalloutText';
export { CalloutTitle } from './Callout/CalloutTitle';
export { Card } from './Card/Card';
export { CardBody } from './Card/CardBody';
export { CardCategory } from './Card/CardCategory';
export { CardFooterCTA } from './Card/CardFooterCTA';
export { CardReadMore } from './Card/CardReadMore';
export { CardSignature } from './Card/CardSignature';
export { CardTag } from './Card/CardTag';
export { CardTagsHeader } from './Card/CardTagsHeader';
export { CardText } from './Card/CardText';
export { CardTitle } from './Card/CardTitle';
export { Chip } from './Chips/Chip';
export { ChipLabel } from './Chips/ChipLabel';
export { Collapse } from './Collapse/Collapse';
export { Dimmer } from './Dimmer/Dimmer';
export { DimmerButtons } from './Dimmer/DimmerButtons';
export { Dropdown } from './Dropdown/Dropdown';
export { DropdownMenu } from './Dropdown/DropdownMenu';
export { DropdownToggle } from './Dropdown/DropdownToggle';
export { FontLoader, useFontLoader } from './FontLoader/FontLoader';
export { Forward } from './Forward/Forward';
export { GoBack } from './GoBack/GoBack';
export { GridItem, GridItemText, GridItemTextWrapper, GridList, GridRow } from './Grid/index';
export { Header } from './Header/Header';
export { HeaderBrand } from './Header/HeaderBrand';
export { HeaderContent } from './Header/HeaderContent';
export { HeaderLinkZone } from './Header/HeaderLinkZone';
export { HeaderRightZone } from './Header/HeaderRightZone';
export { HeaderSearch } from './Header/HeaderSearch';
export { HeaderSocialsZone } from './Header/HeaderSocialsZone';
export { HeaderToggler } from './Header/HeaderToggler';
export { Headers } from './Header/Headers';
export { Hero, HeroBackground, HeroBody, HeroButton, HeroCategory, HeroTitle } from './Hero/index';
export { Icon, clearIconCache, iconsList as icons, preloadIcons } from './Icon/Icon';
export { Input } from './Input/Input';
export { InputContainer } from './Input/InputContainer';
export { TextArea } from './Input/TextArea';
export { LinkList } from './LinkList/LinkList';
export { LinkListItem } from './LinkList/LinkListItem';
export { MegamenuFooter } from './Megamenu/MegamenuFooter';
export { MegamenuHighlightColumn } from './Megamenu/MegamenuHighlightColumn';
export { MegamenuItem } from './Megamenu/MegamenuItem';
export { Modal } from './Modal/Modal';
export { ModalBody } from './Modal/ModalBody';
export { ModalFooter } from './Modal/ModalFooter';
export { ModalHeader } from './Modal/ModalHeader';
export { Nav } from './Nav/Nav';
export { useNavScroll } from './NavScroll/index';
export { NotificationManager, createNotification, notify } from './Notification/index';
export { Pager } from './Pager/Pager';
export { Progress } from './Progress/Progress';
export { Rating } from './Rating/Rating';
export { ResponsiveImage } from './ResponsiveImage/ResponsiveImage';
export { Section } from './Section/Section';
export { Select } from './Select/Select';
export { Sidebar } from './Sidebar/Sidebar';
export { Skiplink } from './Skiplink/Skiplink';
export { SkiplinkItem } from './Skiplink/SkiplinkItem';
export { Spinner } from './Spinner/Spinner';
export { StepperContainer } from './Stepper/StepperContainer';
export { StepperContent } from './Stepper/StepperContent';
export { StepperDots } from './Stepper/StepperDots';
export { StepperHeader } from './Stepper/StepperHeader';
export { StepperHeaderElement } from './Stepper/StepperHeaderElement';
export { StepperNav } from './Stepper/StepperNav';
export { ThumbNav } from './ThumbNav/ThumbNav';
export { ThumbNavItem } from './ThumbNav/ThumbNavItem';
export { TimelinePin } from './Timeline/TimelinePin';
export { Timeline } from './Timeline/TimelineWrapper';
export { Toggle } from './Toggle/Toggle';
export { Toolbar } from './Toolbar/Toolbar';
export { ToolbarDividerItem } from './Toolbar/ToolbarDividerItem';
export { ToolbarItem } from './Toolbar/ToolbarItem';

// Types
export type { AccordionProps } from './Accordion/Accordion';
export type { AccordionBodyProps } from './Accordion/AccordionBody';
export type { AccordionHeaderProps } from './Accordion/AccordionHeader';
export type { AlertProps } from './Alert/Alert';
export type { AvatarContainerProps } from './Avatar/AvatarContainer';
export type { AvatarExtraTextProps } from './Avatar/AvatarExtraText';
export type { AvatarGroupContainerProps } from './Avatar/AvatarGroupContainer';
export type { AvatarIconProps } from './Avatar/AvatarIcon';
export type { AvatarPresenceProps } from './Avatar/AvatarPresence';
export type { AvatarStatusProps } from './Avatar/AvatarStatus';
export type { AvatarWrapperProps } from './Avatar/AvatarWrapper';
export type { BadgeProps } from './Badge/Badge';
export type { BottomNavProps } from './BottomNav/BottomNav';
export type { BottomNavItemProps } from './BottomNav/BottomNavItem';
export type { BreadcrumbProps } from './Breadcrumb/Breadcrumb';
export type { ButtonProps } from './Button/Button';
export type { CalloutProps } from './Callout/Callout';
export type { CalloutTextProps } from './Callout/CalloutText';
export type { CalloutTitleProps } from './Callout/CalloutTitle';
export type { CardProps } from './Card/Card';
export type { CardBodyProps } from './Card/CardBody';
export type { CardCategoryProps } from './Card/CardCategory';
export type { CardFooterCTAProps } from './Card/CardFooterCTA';
export type { CardReadMoreProps } from './Card/CardReadMore';
export type { CardSignatureProps } from './Card/CardSignature';
export type { CardTagProps } from './Card/CardTag';
export type { CardTagsHeaderProps } from './Card/CardTagsHeader';
export type { CardTextProps } from './Card/CardText';
export type { CardTitleProps } from './Card/CardTitle';
export type { ChipProps } from './Chips/Chip';
export type { ChipLabelProps } from './Chips/ChipLabel';
export type { DimmerProps } from './Dimmer/Dimmer';
export type { DimmerButtonsProps } from './Dimmer/DimmerButtons';
export type { DropdownProps } from './Dropdown/Dropdown';
export type { DropdownMenuProps } from './Dropdown/DropdownMenu';
export type { DropdownToggleProps } from './Dropdown/DropdownToggle';
export type { FontLoaderProps } from './FontLoader/FontLoader';
export type { ForwardProps } from './Forward/Forward';
export type {
  GridItemProps,
  GridItemTextProps,
  GridItemTextWrapperProps,
  GridListProps,
  GridRowProps
} from './Grid/index';
export type { HeaderProps } from './Header/Header';
export type { HeaderBrandProps } from './Header/HeaderBrand';
export type { HeaderContentProps } from './Header/HeaderContent';
export type { HeaderLinkZoneProps } from './Header/HeaderLinkZone';
export type { HeaderRightZoneProps } from './Header/HeaderRightZone';
export type { HeaderSearchProps } from './Header/HeaderSearch';
export type { HeaderSocialsZoneProps } from './Header/HeaderSocialsZone';
export type { HeaderTogglerProps } from './Header/HeaderToggler';
export type { HeadersProps } from './Header/Headers';
export type {
  HeroBackgroundProps,
  HeroBodyProps,
  HeroButtonProps,
  HeroCategoryProps,
  HeroProps,
  HeroTitleProps
} from './Hero/index';
export type { IconName, IconProps } from './Icon/Icon';
export type { InputProps } from './Input/Input';
export type { InputContainerProps } from './Input/InputContainer';
export type { TextAreaProps } from './Input/TextArea';
export type { LinkListProps } from './LinkList/LinkList';
export type { LinkListItemProps } from './LinkList/LinkListItem';
export type { MegamenuFooterProps } from './Megamenu/MegamenuFooter';
export type { MegamenuHCProps } from './Megamenu/MegamenuHighlightColumn';
export type { MegamenuItemProps } from './Megamenu/MegamenuItem';
export type { ModalProps } from './Modal/Modal';
export type { ModalBodyProps } from './Modal/ModalBody';
export type { ModalFooterProps } from './Modal/ModalFooter';
export type { ModalHeaderProps } from './Modal/ModalHeader';
export type { NavProps } from './Nav/Nav';
export type { ChangesType, RegisterOptions, useNavScrollArgs, useNavScrollResult } from './NavScroll';
export type { NotificationId, NotificationManagerProps, NotificationOptions, NotificationProps } from './Notification';
export type { PagerProps } from './Pager/Pager';
export type { ProgressProps } from './Progress/Progress';
export type { RatingProps } from './Rating/Rating';
export type { ResponsiveImageProps } from './ResponsiveImage/ResponsiveImage';
export type { SectionProps } from './Section/Section';
export type { SelectProps } from './Select/Select';
export type { SidebarProps } from './Sidebar/Sidebar';
export type { SkiplinkProps } from './Skiplink/Skiplink';
export type { SkiplinkItemProps } from './Skiplink/SkiplinkItem';
export type { SpinnerProps } from './Spinner/Spinner';
export type { StepperContainerProps } from './Stepper/StepperContainer';
export type { StepperContentProps } from './Stepper/StepperContent';
export type { StepperDotsProps } from './Stepper/StepperDots';
export type { StepperHeaderProps } from './Stepper/StepperHeader';
export type { StepperHeaderElementProps } from './Stepper/StepperHeaderElement';
export type { StepperNavProps } from './Stepper/StepperNav';
export type { ThumbNavProps } from './ThumbNav/ThumbNav';
export type { ThumbNavItemProps } from './ThumbNav/ThumbNavItem';
export type { TimelinePinProps } from './Timeline/TimelinePin';
export type { TimelineProps } from './Timeline/TimelineWrapper';
export type { ToggleProps } from './Toggle/Toggle';
export type { ToolbarProps } from './Toolbar/Toolbar';
export type { ToolbarItemBadge, ToolbarItemProps } from './Toolbar/ToolbarItem';

export type {
  BreadcrumbItemProps,
  ButtonDropdownProps,
  ButtonGroupProps,
  ButtonToolbarProps,
  CardColumnsProps,
  CardDeckProps,
  CardFooterProps,
  CardGroupProps,
  CardHeaderProps,
  CardImgOverlayProps,
  CardImgProps,
  CardLinkProps,
  CardSubtitleProps,
  ColProps,
  ContainerProps,
  FadeProps,
  FormFeedbackProps,
  FormGroupProps,
  FormProps,
  FormTextProps,
  InputGroupProps,
  InputGroupTextProps,
  LabelProps,
  ListGroupItemHeadingProps,
  ListGroupItemProps,
  ListGroupItemTextProps,
  ListGroupProps,
  MediaProps,
  NavItemProps,
  NavLinkProps,
  NavbarBrandProps,
  NavbarProps,
  NavbarTogglerProps,
  PaginationItem as PagerItemProps,
  PaginationLink as PagerLinkProps,
  PopoverBodyProps,
  PopoverHeaderProps,
  PopoverProps,
  RowProps,
  TabContentProps,
  TabPaneProps,
  TableProps,
  TooltipProps,
  UncontrolledAlertProps,
  UncontrolledCollapseProps,
  UncontrolledTooltipProps
} from 'reactstrap';
