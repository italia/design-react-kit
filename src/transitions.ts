// Taken from Reactstrap utils.js which are tken from bootstrap sass
export const TransitionTimeouts = {
  // $transition-fade
  Fade: 150,
  // $transition-collapse
  Collapse: 350,
  // $modal-transition
  Modal: 300,
  // $carousel-transition
  Carousel: 600
};

// taken from Reactstrap utils.js
export const TransitionsKeys = [
  'in',
  'mountOnEnter',
  'unmountOnExit',
  'appear',
  'enter',
  'exit',
  'timeout',
  'onEnter',
  'onEntering',
  'onEntered',
  'onExit',
  'onExiting',
  'onExited'
];

const transitionStatus = {
  entering: 1,
  entered: 1,
  exiting: 1,
  exited: 1
};

export type TransitionStates = keyof typeof transitionStatus;
