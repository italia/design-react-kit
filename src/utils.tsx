export const noop = () => {};

export const logError = (message: string) => {
  if (typeof console !== 'undefined') {
    if (console.error) {
      console.error(message);
    } else {
      console.log(message);
    }
  }
};

// @internal Used for testing purposes only
export const flushMessageCache = () => {
  for (const key in messages) {
    delete messages[key];
  }
};

const messages: Record<string, 1> = {};
export const notifyDeprecation = (
  message: string,
  options: { once: boolean } = { once: true }
) => {
  if (!options.once) {
    logError(message);
  } else {
    if (!messages[message]) {
      logError(message);
      messages[message] = 1;
    }
  }
};

export function mapToCssModules(
  className: string | null,
  cssModules?: Record<string, string>
) {
  let finalClassNames = className == null ? '' : className;
  if (!cssModules) {
    return finalClassNames;
  }
  return finalClassNames
    .split(' ')
    .map((klass) => cssModules[klass] ?? klass)
    .join(' ');
}

type UnknownObject = Record<string, unknown>;

export function pick<T extends UnknownObject>(
  obj: T,
  keys: keyof T | Array<keyof T>
): Partial<T> {
  const keysArray = Array.isArray(keys) ? keys : [keys];
  const newObj: Partial<T> = {};
  for (const key of keysArray) {
    if (key in obj) {
      newObj[key] = obj[key];
    }
  }
  return newObj;
}

export function omit<T extends UnknownObject>(
  obj: T,
  keys: keyof T | Array<keyof T>
): Partial<T> {
  const keysLookup = new Set(Array.isArray(keys) ? keys : [keys]);

  const newObj: Partial<T> = {};
  for (const key in obj) {
    if (!keysLookup.has(key)) {
      newObj[key] = obj[key];
    }
  }
  return newObj;
}
