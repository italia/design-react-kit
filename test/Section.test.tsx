import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { Section } from '../src';
import { muteConsoleWithCheck } from './helper';

// Test for breaking changes
test('Should support old neutral prop even as @deprecated', () => {
  const { container } = muteConsoleWithCheck(() => render(<Section neutral />));
  expect(container.firstChild).toHaveClass('section section-neutral');
});

test('Should support old muted prop even as @deprecated', () => {
  const { container } = muteConsoleWithCheck(() => render(<Section muted />));
  expect(container.firstChild).toHaveClass('section section-muted');
});

test('Should support new color prop', () => {
  const { container } = render(<Section color='neutral' />);
  expect(container.firstChild).toHaveClass('section section-neutral');
});

test('Should support old className behaviour with special flag', () => {
  const { container } = render(<Section className='myClass' wrapperClassName={true} />);
  expect(container.firstChild).toHaveClass('myClass');
});

test('Should not apply className to the wrapper', () => {
  const { container } = render(<Section className='myClass' />);
  expect(container.firstChild).not.toHaveClass('myClass');
});

test('Should apply wrapperClassName to the wrapper', () => {
  const { container } = render(<Section wrapperClassName='myClass' />);
  expect(container.firstChild).toHaveClass('myClass');
});

test('Should not set backgroundImage when the image prop is undefined', () => {
  const { container } = render(<Section image={undefined} />);
  expect(container.firstChild).not.toHaveStyle('background-image: url("undefined")');
});

test('should have a testId for resilient UI changes', () => {
  render(<Section testId='test-id-section' />);

  expect(screen.getByTestId('test-id-section')).toBeTruthy();
});

// Write more proper tests here
