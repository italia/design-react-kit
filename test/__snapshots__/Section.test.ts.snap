// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories Snapshots Documentazione/Componenti/Section Esempio 1`] = `
<div>
  <div
    class="section"
  >
    <div
      class="section-content"
    >
      <div
        class="container"
      >
        <div
          class="mb-3 row"
        >
          <div
            class="col-12"
          >
            <h4>
              Morbi fermentum amet
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="pe-0 pe-md-5 mb-3 col-12 col-lg-6 col-xl-4"
          >
            Platea dictumst vestibulum rhoncus est pellentesque elit ullamcorper dignissim cras. Dictum sit amet justo donec enim diam vulputate ut. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="pe-0 pe-md-5 mb-3 col-12 col-lg-6 col-xl-4"
          >
            Eget egestas purus viverra accumsan. Diam maecenas ultricies mi eget mauris pharetra et. Etiam dignissim diam quis enim. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="pe-0 pe-md-5 mb-3 col-12 col-lg-6 col-xl-4"
          >
            Euismod lacinia at quis risus sed vulputate. Scelerisque purus semper eget duis at tellus at urna condimentum. Mattis enim ut tellus elementum sagittis.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Section ImmagineDiSfondo 1`] = `
<div>
  <div
    class="section section-image"
    style="background-image: url(https://picsum.photos/1280/720?image=811);"
  >
    <div
      class="section-content"
    >
      <div
        class="white-color container"
      >
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            <h4>
              Morbi fermentum amet
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Platea dictumst vestibulum rhoncus est pellentesque elit ullamcorper dignissim cras. Dictum sit amet justo donec enim diam vulputate ut. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Eget egestas purus viverra accumsan. Diam maecenas ultricies mi eget mauris pharetra et. Etiam dignissim diam quis enim. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Euismod lacinia at quis risus sed vulputate. Scelerisque purus semper eget duis at tellus at urna condimentum. Mattis enim ut tellus elementum sagittis.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Section SectionConCard 1`] = `
<div>
  <div
    class="section section-muted"
  >
    <div
      class="section-content"
    >
      <h4>
        Morbi fermentum amet
      </h4>
      <div
        class="row"
      >
        <div
          class="col-sm-6"
        >
          <div
            class="shadow h-100 card"
          >
            <div
              class="card-body"
            >
              <p
                class="card-text"
              >
                Platea dictumst vestibulum rhoncus est pellentesque elit ullamcorper dignissim cras. Dictum sit amet justo donec enim diam vulputate ut. Eu nisl nunc mi ipsum faucibus. Eget egestas purus viverra accumsan. Diam maecenas ultricies mi eget mauris pharetra et.
                 
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-sm-6"
        >
          <div
            class="shadow h-100 card"
          >
            <div
              class="card-body"
            >
              <p
                class="card-text"
              >
                Etiam dignissim diam quis enim. Eu nisl nunc mi ipsum faucibus. Euismod lacinia at quis risus sed vulputate. Scelerisque purus semper eget duis at tellus at urna condimentum. Mattis enim ut tellus elementum sagittis.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Section SfondoNeutrale 1`] = `
<div>
  <div
    class="section section-neutral"
  >
    <div
      class="section-content"
    >
      <div
        class="white-color container"
      >
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            <h4>
              Morbi fermentum amet
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Platea dictumst vestibulum rhoncus est pellentesque elit ullamcorper dignissim cras. Dictum sit amet justo donec enim diam vulputate ut. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Eget egestas purus viverra accumsan. Diam maecenas ultricies mi eget mauris pharetra et. Etiam dignissim diam quis enim. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Euismod lacinia at quis risus sed vulputate. Scelerisque purus semper eget duis at tellus at urna condimentum. Mattis enim ut tellus elementum sagittis.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Section SfondoPrimario 1`] = `
<div>
  <div
    class="section section-primary"
  >
    <div
      class="section-content"
    >
      <div
        class="white-color container"
      >
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            <h4>
              Morbi fermentum amet
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Platea dictumst vestibulum rhoncus est pellentesque elit ullamcorper dignissim cras. Dictum sit amet justo donec enim diam vulputate ut. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Eget egestas purus viverra accumsan. Diam maecenas ultricies mi eget mauris pharetra et. Etiam dignissim diam quis enim. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Euismod lacinia at quis risus sed vulputate. Scelerisque purus semper eget duis at tellus at urna condimentum. Mattis enim ut tellus elementum sagittis.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Section SfondoTenue 1`] = `
<div>
  <div
    class="section section-muted"
  >
    <div
      class="section-content"
    >
      <div
        class="container"
      >
        <div
          class="mb-3 row"
        >
          <div
            class="col-12"
          >
            <h4>
              Morbi fermentum amet
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Platea dictumst vestibulum rhoncus est pellentesque elit ullamcorper dignissim cras. Dictum sit amet justo donec enim diam vulputate ut. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Eget egestas purus viverra accumsan. Diam maecenas ultricies mi eget mauris pharetra et. Etiam dignissim diam quis enim. Eu nisl nunc mi ipsum faucibus.
          </div>
          <div
            class="col-12 col-lg-6 col-xl-4"
          >
            Euismod lacinia at quis risus sed vulputate. Scelerisque purus semper eget duis at tellus at urna condimentum. Mattis enim ut tellus elementum sagittis.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
